python-avd/pyavd/_eos_designs/structured_config/network_services/utils_wan.py

```diff
@@ -171,35 +175,28 @@ def _update_policy_match_statements(self: AvdStructuredConfigNetworkServicesProt
             required=True,
             custom_error_msg=f"wan_virtual_topologies.policies[{policy['profile_prefix']}].default_virtual_toplogy.",
         )
+        # TODO: this should be removed in a future refactor
+        avt_object = EosDesigns.WanVirtualTopologies.PoliciesItem.DefaultVirtualTopology._from_dict(default_virtual_topology)
         # Separating default_match as it is used differently
         default_match = None
-        if not get(default_virtual_topology, "drop_unmatched", default=False):
-            name = get(
-                default_virtual_topology,
-                "name",
-                default=self._default_profile_name(policy["profile_prefix"], "DEFAULT"),
-            )
+        if not avt_object.drop_unmatched:
+            name = avt_object.name or self._default_profile_name(policy["profile_prefix"], "DEFAULT")
             context_path = f"wan_virtual_topologies.policies[{policy['profile_prefix']}].default_virtual_topology"
             # Verify that path_groups are set or raise
-            get(
-                default_virtual_topology,
-                "path_groups",
-                required=True,
-                custom_error_msg=f"Either 'drop_unmatched' or 'path_groups' must be set under '{context_path}'.",
-            )
+            if not avt_object.path_groups:
+                msg = (f"Either 'drop_unmatched' or 'path_groups' must be set under '{context_path}'.",)
+                raise AristaAvdInvalidInputsError(msg)
             load_balance_policy_name = self.shared_utils.generate_lb_policy_name(name)
-            load_balance_policy = self._generate_wan_load_balance_policy(load_balance_policy_name, default_virtual_topology, context_path)
+            load_balance_policy = self._generate_wan_load_balance_policy(load_balance_policy_name, avt_object, context_path)
             if not load_balance_policy:
                 msg = (
                     f"The `default_virtual_topology` path-groups configuration for `wan_virtual_topologies.policies[{policy['name']}]` produces "
                     "an empty load-balancing policy. Make sure at least one path-group present on the device is allowed in the "
                     "`default_virtual_topology` path-groups."
                 )
                 raise AristaAvdError(msg)
-            application_profile = get(default_virtual_topology, "application_profile", default="default")
-
             default_match = {
-                "application_profile": application_profile,
+                "application_profile": "default",
```

Comment:
for reviewer - this is fixing a bug - application_profile does not exist under default_virtual_topology